### PR TITLE
abac custom function support

### PR DESCRIPTION
**Pull request by slvishal**
#This change support abac custom function policy.

`
p,developer,/api/main,*
p2,"customFunction(r2.sub.key,r2.sub.ip,'execute')",/api/main,POST
`

### CODE REVIEW

**Comment by hsluoyz** [comment]
@jessecooper @dfresh613

File: casbin_couchbase_adapter/adapter.py [hunk 1]
```diff
@@ -17,6 +17,26 @@
 from couchbase.n1ql import STATEMENT_PLUS
 
 
+
+def load_policy_line(tokens, model):
```
Comment (slvishal) [inline casbin_couchbase_adapter/adapter.py:21]: casbin default Adapter is FileAdapter. It reads line by line code each line is a string. Where as couchbase store data in a list. This above change check if token is a list or string.

**Comment by jessecooper** [comment]
@favipcj

**Comment by favipcj** [comment]
I think we can close this PR for now, as it may require some other changes in some dependencies. But once this proposal is ready with some unit tests included, we could open a new PR.

Are you ok with that @slvishal ? or do you want to continue with this PR and add some unit tests and documentation/examples?

**Comment by slvishal** [comment]
We are not using ABAC in flask-authz for our current scenario  i am closing this PR

**Pull request closed without merging**